### PR TITLE
Reworked constructor/defaults

### PR DESCRIPTION
**Pull request by Patitotective**
This PR removes the need for calling `implDefaults` after the `defaults` macro by emitting the procedure in the type definition.
This solves a bug where some object definitions didn't match perfectly in the `defaultTable`.
It also adds the `defBothConstr` flag to define both the normal and the type constructors.

### CODE REVIEW

File: src/constructor/defaults.nim [hunk 1]
```diff
@@ -61,26 +62,21 @@ macro defaults*(tdef: untyped): untyped =
...
-    routine.insertIdentDef 0, identDefTyp("_", routine.returnType.makeTypeDesc())
-    let name =
-      if result[0].strVal.startsWith("new"):
-        ident"new"
-      else:
-        ident"init"
-    result[0] = name
-  if defExported in genFlags:
-    result[0] = result[0].postfix("*")
+  let newProc = newProc(if defExported in genFlags: procName.postfix("*") else: procName, params, objCStr) # Initialization procedure
```
Comment (beef331) [inline src/constructor/defaults.nim]: I'd prefer these if expressions to be on their own lines. Inline if expressions are terrible for readability.

File: src/constructor/defaults.nim [hunk 2]
```diff
@@ -61,26 +62,21 @@ macro defaults*(tdef: untyped): untyped =
...
+
+  params.add newIdentDefs(ident("_"), newTree(nnkBracketExpr, ident("typedesc"), innerIdent))
+
+  let newTypeProc = newProc(if defExported in genFlags: typeProcName.postfix("*") else: typeProcName, params, objCStr) # Initialization procedure
```
Comment (beef331) [inline src/constructor/defaults.nim]: Ditto.

File: src/constructor/defaults.nim (hunk 1 above)
Comment (Patitotective) [reply src/constructor/defaults.nim]: Done :]

File: src/constructor/defaults.nim (hunk 2 above)
Comment (Patitotective) [reply src/constructor/defaults.nim]: Done :]

**Pull request merged**